### DISCUSSION TITLE
Add support for buzzer

### DISCUSSION THREAD

**Discussion by xtimmy86x** [HASwitchPlate/openHASP#290] [Ideas]
Hello, i'm using this project with a arduitouch that have a integrated buzzer connected to a GPIO of the esp32 board 

It's very interesting if it will be integrated and controllable with the home assistant integration

**Comment by nagyrobi** [comment]
Add it as a switch.

**Comment by xtimmy86x** [comment]
i did it but no sound, only a soft crackling sound when i put on and off the switch

**Comment by fvanroie** [comment]
You need to drive the pitch/length via a library. Take a look at the documentation on how to add your [custom code](https://www.openhasp.com/0.7.0/firmware/compiling/custom/).
Make sure that the code is non-blocking or it will interfere with the performance of the UI and other compoments.

**Comment by nagyrobi** [comment]
> i did it but no sound, only a soft crackling sound when i put on and off the switch

You're talking about a piezo speaker, not a buzzer. Likely you picked the wrong part.

A buzzer is able to output a sound (predefined from factory) when some DC is applied to its terminals.

A piezo speaker is just a part of the buzzer. It needs some audio signal to be generated and applied to it to act like a buzzer. Openhasp doesn't support that.

**Reply by adorobis** [reply]
I would be interested to add such buzzer to my screens so that I get a heptic feedback when touching the screen. This could also be a coin vibration motor. Is there a way to control such GPIO output directly on the device or I need to send an MQTT command to control a GPIO output?

**Reply by fvanroie** [reply]
You can check out [`feedback_cb`](https://docs.lvgl.io/7.11/porting/indev.html#other-features) in `lv_indev_drv_t`, which facilitates _playing a sound on `LV_EVENT_CLICK`_. This should also make it possible to vibrate a motor.

**Reply by nagyrobi** [reply]
Wow, this would worth to be exposed to UI, to be used in a "group" with a GPO pin...

**Comment by NdK73** [comment]
I see this is an old discussion, but can't find anything newer.
Did something change in the last year? Maybe RTTTL support (that fits quite nicely in a MQTT message...)?

**Comment by flexiti** [comment]
add librrary , eg. ezBuzzer

**Comment by sohnvonoff** [comment]
Acoustic feedback is very useful for touch input. It is sorely lacking in OpenHASP!
Otherwise it's really good for various small touch displays in conjunction with Homeassistant or other MQTT environments. The programming implementation is great.
